samples/04.1-file-transfer-listener/file-transfer-listener-integration-tests/src/test/java/org/eclipse/dataspaceconnector/extension/sample/test/FileTransferListenerSampleTest.java

```diff
@@ -0,0 +1,102 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.io.File;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.extension.sample.test.FileTransferSampleTestCommon.getFileFromRelativePath;
+
+@EndToEndTest
+public class FileTransferListenerSampleTest {
+
+    static final String CONSUMER_CONFIG_PROPERTIES_FILE_PATH = "samples/04.1-file-transfer-listener/consumer/config.properties";
+    static final String PROVIDER_CONFIG_PROPERTIES_FILE_PATH = "samples/04.0-file-transfer/provider/config.properties";
+    // Reuse an already existing file for the test. Could be set to any other existing file in the repository.
+    static final String SAMPLE_ASSET_FILE_PATH = "samples/04.1-file-transfer-listener/README.md";
+    static final String DESTINATION_FILE_PATH = "samples/04.1-file-transfer-listener/requested.test.txt";
+    // marker.txt is a fixed name and always in the same directory as the file defined with DESTINATION_FILE_PATH.
+    static final String MARKER_FILE_PATH = "samples/04.1-file-transfer-listener/marker.txt";
+    static final File MARKER_FILE = getFileFromRelativePath(FileTransferListenerSampleTest.MARKER_FILE_PATH);
+    static final String MARKER_FILE_CONTENT = "Transfer complete";
+    @RegisterExtension
+    static EdcRuntimeExtension provider = new EdcRuntimeExtension(
+            ":samples:04.0-file-transfer:provider",
+            "provider",
+            Map.of(
+                    // Override 'edc.samples.04.asset.path' implicitly set via property 'edc.fs.config'.
+                    "edc.samples.04.asset.path", getFileFromRelativePath(SAMPLE_ASSET_FILE_PATH).getAbsolutePath(),
+                    "edc.fs.config", getFileFromRelativePath(PROVIDER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    @RegisterExtension
+    static EdcRuntimeExtension consumer = new EdcRuntimeExtension(
+            ":samples:04.1-file-transfer-listener:consumer",
+            "consumer",
+            Map.of(
+                    "edc.fs.config", getFileFromRelativePath(CONSUMER_CONFIG_PROPERTIES_FILE_PATH).getAbsolutePath()
+            )
+    );
+    private final FileTransferSampleTestCommon testUtils = new FileTransferSampleTestCommon(SAMPLE_ASSET_FILE_PATH, DESTINATION_FILE_PATH);
+
+    /**
+     * Run all sample steps in one single test.
+     * Note: Sample steps cannot be separated into single tests because {@link EdcRuntimeExtension}
+     * runs before each single test.
+     */
+    @Test
+    void runSampleSteps() throws Exception {
+        assertTestPrerequisites();
+
+        testUtils.initiateContractNegotiation();
+        testUtils.lookUpContractAgreementId();
+        testUtils.requestTransferFile();
+        testUtils.assertDestinationFileContent();
+        testUtils.assertFileContent(MARKER_FILE, MARKER_FILE_CONTENT);
```

Comment:
nit: this check is specific to sample 4.1 so it can be kept locally here in this test file and not in the test fixture.